File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
temp = []
alp = ['A', 'E', 'I', 'O', 'U']

num = 0

def bfs(word):
    global num

    tmp_word = ''.join(map(str, temp))
    if word == tmp_word:
        return num
    if len(temp) == 5:
        return

    for a in alp:
        num += 1
        temp.append(a)
        if bfs(word):
            return num
        temp.pop()


def solution(word):
    bfs(word)
    return num

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

vector<char> temp;
vector<char> alp = {'A', 'E', 'I', 'O', 'U'};
int num = 0;

int bfs(string word) {
    string tmp_word(temp.begin(), temp.end());
    if (word == tmp_word) {
        return num;
    }
    if (temp.size() == 5) {
        return -1;  // Equivalent to Python's "return"
    }

    for (char a : alp) {
        num++;
        temp.push_back(a);
        int result = bfs(word);
        if (result != -1) {
            return result;
        }
        temp.pop_back();
    }

    return -1;
}

int solution(string word) {
    int result = bfs(word);
    return result;
}
